script.sh

```diff
@@ -1,36 +1,67 @@
 #!/bin/bash
 set -
 
-DOWNLOAD_URL="https://www.supermicro.com/wdl/utility/IPMIView/Linux/"
-LOCAL_DOWNLOAD_LOCATION="./SM_download"
-
-if which wget >/dev/null; then
-  echo "Downloading latest version of IPMIView from [${DOWNLOAD_URL}]..."
-  wget \
-    --timestamping \
-    --recursive \
-    --level=1 \
-    -q \
-    --show-progress \
-    --directory-prefix="${LOCAL_DOWNLOAD_LOCATION}/" \
-    --no-parent \
-    --no-directories \
-    --reject index.html,index.html.tmp,robots.txt,robots.txt.tmp \
-    "${DOWNLOAD_URL}"
-
-  # Check SHA-256
-  EXPECTED_SHA256=$(\grep -A3 "tar.gz" "${LOCAL_DOWNLOAD_LOCATION}/CheckSum.txt" | grep SHA-256 | cut -d':' -f2 | tr -d "[:space:]" | tr '[:upper:]' '[:lower:]')
-  ACTUAL_SHA256=$(shasum -a 256 "${LOCAL_DOWNLOAD_LOCATION}"/IPMIView*.tar* | cut -d' ' -f1 | tr -d "[:space:]" | tr '[:upper:]' '[:lower:]')
-  if ! diff <(echo "${EXPECTED_SHA256}") <(echo "${ACTUAL_SHA256}"); then
-    echo "SHA-256 is not as expected; download corrupted."
-    echo "Expected: [${EXPECTED_SHA256}]"
-    echo "Actual:   [${ACTUAL_SHA256}]"
+BASE_URL="https://www.supermicro.com/en/support/resources"
+INFO_URL="${BASE_URL}/downloadcenter/smsdownload\?category\=IPMI"
+API_URL="https://www.supermicro.com/support/resources/getfile.php?SoftwareItemID=DLID&type=serversoftwarefile"
+
+if which curl >/dev/null; then
+  # this area is subject to problems if SuperMicro changes their page format
```

Comment:
Thanks for adding this comment; I agree, this approach seems more brittle than the previous approach. Do you think it makes sense to keep the prior approach as fallback logic in case this does not work as expected?

Reply:
good point.  I added some code to fall back to this latest known version, along with a message